File: C++/Best-Time-to-Buy-and-Sell-Stock-II.cpp
/*Runtime: 8 ms, faster than 96.30% of C++ online submissions for Best Time to Buy and Sell Stock II.
Memory Usage: 13.4 MB, less than 5.04% of C++ online submissions for Best Time to Buy and Sell Stock II.
*/

class Solution
{
	public:
		int maxProfit(vector<int> &prices)
		{
			int sum = 0;
			int count =sum;
			//If the number of element in the array are zero or one just return zero.
			if (prices.size() == 1 or prices.size() == 0)
			{
				return 0;
			}

			//Traverse the array and compare the consecutive two elements .
			for (int i = 0; i < prices.size() - 1; i++)
			{
				// If first consecuitve element is less than second subtract both and add in the sum varibale. 
				if (prices[i] < prices[i + 1])
				{
					sum += prices[i + 1] - prices[i];

				}
			}

			//Finally return the sum.
			return sum;

		}
};

File: Python/best-time-to-buy-and-sell-stock-ii.py
'''
Speed: 99.14%
Memory:
Time Complexity: O(n)
'''
class Solution:
    def solve(self, prices):
        # Write your code here
        if len(prices)<2:return 0
        buy = prices[0]
        profit = 0
        for i in range(1,len(prices)):
            if prices[i]<prices[i-1]:
                profit+=prices[i-1]-buy
                buy = prices[i]
        profit+=prices[-1]-buy
        return profit